functests/2_performance_update/kubelet.go

```diff
@@ -0,0 +1,170 @@
+		workerRTNodes, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+
+		workerRTNodes, err = nodes.MatchingOptionalSelector(workerRTNodes)
+		Expect(err).ToNot(HaveOccurred())
+
+		profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+
+		performanceMCP, err = mcps.GetByProfile(profile)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Verify that worker and performance MCP have updated state equals to true
+		for _, mcpName := range []string{testutils.RoleWorker, performanceMCP} {
+			mcps.WaitForCondition(mcpName, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+		}
+
+	})
+	BeforeEach(func() {
+		if discovery.Enabled() && testutils.ProfileNotFound {
+			Skip("Discovery mode enabled, performance profile not found")
+		}
+	})
+	Context("Additional kubelet arguments", func() {
+		It("[test_id:45488]Test performance profile annotation for changing multiple kubelet settings", func() {
+			sysctlAnnotations := map[string]string{
+				"kubeletconfig.experimental": "{\"allowedUnsafeSysctls\":[\"net.core.somaxconn\",\"kernel.msg*\"],\"systemReserved\":{\"memory\":\"300Mi\"},\"kubeReserved\":{\"memory\":\"768Mi\"},\"imageMinimumGCAge\":\"3m\"}",
+			}
+			profile.SetAnnotations(sysctlAnnotations)
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			profiles.UpdateWithRetry(profile)
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+			By("Waiting when mcp finishes updates")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+			kubeletArguments := []string{"/bin/bash", "-c", "ps -ef | grep kubelet | grep config"}
+			for _, node := range workerRTNodes {
+				kubeletConfig, err := nodes.GetKubeletConfig(&node)
+				Expect(err).ToNot(HaveOccurred())
+				sysctlsValue := kubeletConfig.AllowedUnsafeSysctls
+				Expect(kubeletSliceValues(sysctlsValue, "net.core.somaxconn")).To(BeTrue())
+				Expect(kubeletSliceValues(sysctlsValue, "kernel.msg*")).To(BeTrue())
+				Expect(kubeletConfig.KubeReserved["memory"]).To(Equal("768Mi"))
+				stdout, err := nodes.ExecCommandOnNode(kubeletArguments, &node)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(strings.Contains(stdout, "300Mi")).To(BeTrue())
+				Expect(kubeletConfig.ImageMinimumGCAge.Seconds()).To(Equal(180))
+			}
+		})
+		It("[test_id:45493]Kubelet Config is not reflected on remote worker node", func() {
+			cpuManagerAnnotation := map[string]string{
+				"kubeletconfig.experimental": "{\"cpuManagerPolicy\":\"static\",\"cpuManagerReconcilePeriod\":\"5s\"}",
+			}
+			profile.SetAnnotations(cpuManagerAnnotation)
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			profiles.UpdateWithRetry(profile)
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
```

Comment:
check for error

Reply:
Since mcps.WaitForCondition doesn not return anything, it can't be wrapped using Expect.